### PR TITLE
dont order partitions if there are none

### PR DESCRIPTION
**Pull request by vwaurich**

### CODE REVIEW

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2017-05-04_14-21-17](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4248/console).

**Comment by casella** [comment]
Thanks! I have been waiting for this for a long time :)

File: Compiler/BackEnd/SynchronousFeatures.mo [hunk 1]
```diff
@@ -673,15 +673,16 @@ algorithm
...
+  if (not listEmpty(outSysts)) then
+	  //order the partitions according to their causality order and corresponding subclock order
+	  outSysts_noOrder := arrayCreate(listLength(outSysts), listHead(outSysts));
+	  for i in List.intRange(arrayLength(order)) loop
+	    outSys := listGet(outSysts,i);
+	    outSys.partitionKind := BackendDAE.CLOCKED_PARTITION(order[i] + off);
+	    arrayUpdate(outSysts_noOrder, order[i], outSys);
+	  end for;
+	  outSysts := arrayList(outSysts_noOrder);
+	end if;
```
Comment (sjoelund) [inline Compiler/BackEnd/SynchronousFeatures.mo]: Please don't use tabs... The pre-commit hook even fixes this (if using Linux)

Comment (vwaurich) [inline Compiler/BackEnd/SynchronousFeatures.mo]: Sorry, I was relying on AnyEdit tab conversion in eclipse but it lost its settings.

**Pull request closed without merging**